internal/kubernetes/operator/config_exporter.go

```diff
@@ -0,0 +1,167 @@
+	// TODO: Add REST to OPERATOR entities matcher
+
+	output := bytes.NewBufferString(yamlSeparator)
+	var resources []runtime.Object
+
+	serializer := json.NewSerializerWithOptions(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme,
+		json.SerializerOptions{Yaml: true, Pretty: true})
+
+	projectResources, projectName, err := e.exportProject()
+	if err != nil {
+		return "", err
+	}
+	resources = append(resources, projectResources...)
+
+	deploymentsResources, err := e.exportDeployments(projectName)
+	if err != nil {
+		return "", err
+	}
+	resources = append(resources, deploymentsResources...)
+
+	for _, res := range resources {
+		err = serializer.Encode(res, output)
+		if err != nil {
+			return "", err
+		}
+		output.WriteString(yamlSeparator)
+	}
+
+	return output.String(), nil
+}
+
+func (e *ConfigExporter) exportProject() ([]runtime.Object, string, error) {
+	var resources []runtime.Object
+
+	// Project
+	projectData, relatedSecrets, err := project.BuildAtlasProject(e.dataProvider, e.projectID, e.targetNamespace, e.includeSecretsData)
+	if err != nil {
+		return nil, "", err
+	}
+	resources = append(resources, projectData)
+	for _, secret := range relatedSecrets {
+		resources = append(resources, secret)
+	}
+
+	// Project secret with credentials
+	resources = append(resources, project.BuildProjectConnectionSecret(e.credsProvider,
+		projectData.Name,
+		projectData.Namespace, e.orgID, e.includeSecretsData))
+
+	// DB users
+	usersData, relatedSecrets, err := dbusers.BuildDBUsers(e.dataProvider, e.projectID, projectData.Name, e.targetNamespace, e.includeSecretsData)
+	if err != nil {
+		return nil, "", err
+	}
+	for _, user := range usersData {
+		resources = append(resources, user)
+	}
+	for _, secret := range relatedSecrets {
+		resources = append(resources, secret)
+	}
+
+	return resources, projectData.Name, nil
+}
+
+func (e *ConfigExporter) exportDeployments(projectName string) ([]runtime.Object, error) {
+	var result []runtime.Object
+
+	for _, deploymentName := range e.clusters {
+		// Try advanced cluster first
+		advancedCluster, err := deployment.BuildAtlasAdvancedDeployment(e.dataProvider, e.projectID, projectName, deploymentName, e.targetNamespace)
+		if err == nil {
+			// Append deployment to result
+			result = append(result, advancedCluster.Deployment)
+			// Append backup schedule
+			if advancedCluster.BackupSchedule != nil {
+				result = append(result, advancedCluster.BackupSchedule)
+			}
+			// Append backup policies (one)
+			for _, policy := range advancedCluster.BackupPolicies {
+				if policy != nil {
+					result = append(result, policy)
+				}
+			}
+			continue
+		}
+
+		// Try serverless cluster next
+		serverlessCluster, err := deployment.BuildServerlessDeployments(e.dataProvider, e.projectID, deploymentName, e.targetNamespace)
+		if err == nil {
+			result = append(result, serverlessCluster)
+			continue
+		}
+		return nil, fmt.Errorf("cluster %s not found for project %s", deploymentName, e.projectID)
```

Comment:
let's properly type the errors when doing `fmt.Errorf`
```go
var ErrClusterNotFound = errors.New("cluster not found")
```
```suggestion
		return nil, fmt.Errorf("%w: %s(%s)", ErrClusterNotFound, deploymentName, e.projectID)
```
(copy up to you important bit wrapping the error)

Reply:
Thanks! Fixed